llama_stack/apis/post_training/post_training.py

```diff
@@ -184,7 +184,7 @@ async def supervised_fine_tune(
             description="Model descriptor from `llama model list`",
         ),
         checkpoint_dir: Optional[str] = None,
-        algorithm_config: Optional[AlgorithmConfig] = None,
+        algorithm_config: Optional[Union[LoraFinetuningConfig, QATFinetuningConfig]] = None,
```

Comment:
The idiomatic python way to type this now is:
```suggestion
        algorithm_config: LoraFinetuningConfig | QATFinetuningConfig | None = None,
```

Reply:
Maybe this construction doesn't work w/ Python 3.10 though (our min version).

Reply:
I think it was added in 3.10 actually, so we can apply this.